test/test_rest_utils.py

```diff
@@ -132,3 +134,26 @@ def testSetContentDisposition(name, disp, expected):
         assert rest.setContentDisposition(name, setHeader=False) == expected
     else:
         assert rest.setContentDisposition(name, disp, setHeader=False) == expected
+
+
+@pytest.mark.parametrize('allowed,origin,expected', [
+    ('*', 'thing.com', '*'),
+    ('*,thing.com', 'thing.com', 'thing.com'),
+    ('thing.com', 'thing.com', 'thing.com'),
+    ('thing.com', 'notthing.com', None),
+    ('thing.com', 'subdomain.thing.com', None),
+    ('*.thing.com', 'subdomain.thing.com', 'subdomain.thing.com'),
+    (
+        'fixed*more-fixed.thing.com',
+        'fixed-something-more-fixed.thing.com',
+        'fixed-something-more-fixed.thing.com'
+    ),
+    ('fixed*more-fixed.thing.com', 'other-fixed*other.thing.com', None),
```

Comment:
Not sure what the point of this particular case is. An actual origin shouldn't typically have a wildcard. If you want to test what happens if the origin did have a `*`,  it seems like the case could be simpler.

Reply:
I believe the two tests above your comment are testing against the case I described in the PR description. Specifically, putting a wildcard in the middle of the subdomain of the origin. I know this test case is technically identical to the one on line 145, but I figured I'd include it anyway.